LanguageFeatures/Patterns/Exhaustiveness/exhaustiveness_sealed_A03_t01.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Switch statements and expressions with a sealed class as a
+/// matched type are always exhaustive
+///
+/// @description Check that if a matched value of a switch statement is not a
+/// sealed class then flow analysis treats this statement as a not exhaustive
```

Comment:
I think being 'not a sealed class' is much too broad here. For instance, an enum value is definitely not a sealed class, but covering all values of an enum type can be highly relevant to exhaustiveness.

Again, I think the direct way to check that the compiler/analyzer treats a scrutinee as having a non-exhaustive type is to have a switch statement which isn't exhaustive. That's OK if the scrutinee type is not always-exhaustive, but it is an error if it is always-exhaustive.